### PR TITLE
Microsoft.PolicyInsights/remediations 2019-07-01

### PR DESCRIPTION
**Pull request by pilor**
Adds the resourceDiscoveryMode enum to the remediations API in Microsoft.PolicyInsights.  This allows remediations to discover non-compliant resources and remediate them in one call (optionally).  Previously, non-compliant resources had to be known prior to remediation starting.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by majastrz — APPROVED** [review]
signed off from ARM side.

**Review by sandipsh — APPROVED** [review]
:shipit:

**Comment by pilor** [comment]
@MyronFanQiu Looks like an SDK reviewer didn't get assigned to this PR.  Can you assign someone appropriate?

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-07-01/examples/Remediations_CancelResourceGroupScope.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "parameters": {
+    "subscriptionId": "35ee058e-5fa0-414c-8145-3ebb8d09b6e2",
+    "resourceGroupName": "rg1",
```
Comment (fengzhou-msft) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-07-01/examples/Remediations_CancelResourceGroupScope.json]: Can you change it to `myResourceGroup` for resourceGroupName example values, which is the conventional name used in examples.

**Comment by pilor** [comment]
@fengzhou-msft comments resolved

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**